### PR TITLE
Made gradient descent set its learning rate as min(sigma0)

### PR DESCRIPTION
**Pull request by MichaelClerx**
Updated its tests to ignore boundaries. Updated its docstring to make clear that it does this.

### CODE REVIEW

**Comment by MichaelClerx** [comment]
Again now a new one @DavAug ! But did notice some things while copying its tests

**Review by DavAug — APPROVED** [review]
Thanks @MichaelClerx I am happy with the changes. I am just wondering whether it makes sense to link the learning rate to sigma0. Don't we use sigma0 to randomise the initial seeding of the optimiser which is not really related to the learning rate? So, it might be cleaner to simply separate sigma0 and the learning rate, eta, entirely and introduce eta as an input parameter to the method with a default value?

**Comment by MichaelClerx** [comment]
Thanks @DavAug 
We can't really have default values other than x0 and sigma0, because the controller is responsible for creating the optimisers most of the time. The users can already set the learning rate via the controller (`.optimiser().set_learning_rate()`) & i think that's the best we can do

Regarding the link to `sigma0`. It's used by stochastic optimisers as a "standard deviation", which tells you about the "scale" of the parameters i.e. the sensitivity of the error/logpdf w.r.t. those parameters (at x0). In this (deterministic) method the learning rate is related to that scale, so it made sense to me to set a first guess based on sigma?
